### PR TITLE
Private routes, auth

### PR DESCRIPTION
**Pull request by ranitasahaa1230**
Added private routes to cart, wishlist, and single product page
Added user profile page

### CODE REVIEW

File: src/pages/UserProfile/UserProfile.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  outline: none;
+  color: var(--bg-code);
+  border-radius: 0.8rem;
+  font-size: 1.8rem;
+}
+.logout__btn:hover{
+    opacity: 0.9;
+}
+
+@media (max-width: 645px) {
```
Comment (prakashsakari) [inline src/pages/UserProfile/UserProfile.css:57]: you can use em/rem as the unit over px

File: src/pages/UserProfile/UserProfile.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    .box__page{
+        width: 45rem;
+    }
+  }
+
+  @media (max-width: 378px) {
```
Comment (prakashsakari) [inline src/pages/UserProfile/UserProfile.css:63]: here as well

File: src/pages/Auth/SignUp.js [hunk 3]
```diff
@@ -87,32 +96,73 @@ export function SignUp() {
...
-          }
+            <input
+              type={showPassword ? "text" : "password"}
+              className="visibility-fields"
+              placeholder="Enter Password"
+              value={password}
+              onChange={(e) =>
+                dispatch({ type: "PASSWORD", payload: e.target.value })
+              }
+              autoComplete="off"
```
Comment (Tanay-J) [inline src/pages/Auth/SignUp.js:117]: got to know this attribute

File: src/component/Navbar/Navbar.js [hunk 4]
```diff
@@ -2,21 +2,25 @@ import React from "react";
...
     cartState: { cart, wishList },
     dispatch,
   } = useData();
   const { pathname } = useLocation();
+  const {
+    state: { isAuth, user },
+  } = useAuth();
 
   return (
     <div>
```
Comment (Tanay-J) [inline src/component/Navbar/Navbar.js]: this div could be removed

File: src/reducers/signupReducer.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const signupReducer = (state, action) => {
```
Comment (prakashsakari) [inline src/reducers/signupReducer.js:1]: you can destructure type and payload directly instead of writing action

**Review by prakashsakari — APPROVED** [review]
UI looks stunning
Design is neat and clean
Have added few check comments, you can have a look at them
Looks good to be merged

File: src/pages/Auth/Login.js [hunk 6]
```diff
@@ -13,37 +13,34 @@ export function Login() {
...
-      //   payload: foundUser.wishlist,
-      // });
-
+      const response = await axios.post("/api/auth/login", loginForm);
+      const { foundUser: user, encodedToken } = response.data;
+      dispatch({ type: "AUTH_SUCCESS", payload: { user, encodedToken } });
+      localStorage.setItem("user", JSON.stringify(user));
       localStorage.setItem("token", encodedToken);
-      navigate("/");
+      if (location.state !== null) {
```
Comment (Tanay-J) [inline src/pages/Auth/Login.js:30]: could have used ternary operator here to be consistent with ES6 usage

**Review by Tanay-J — APPROVED** [review]
Good work at keeping the code clean. Good use of context and reducer. You may merge

**Pull request merged**